app/controllers/vm_common.rb

```diff
@@ -1051,6 +1055,26 @@ def scan_histories
     end
   end
 
+  def parent_folder_id(vm)
```

Comment:
`vm.parent_folder` is expensive. Would be nice to avoid that.

Think this is determining the html/css id for the parent rather than the id of the parent folder / cluster.